src/Queue/QueueBase.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\redis\Queue\QueueBase.
+ */
+
+namespace Drupal\redis\Queue;
+
+use Drupal\Core\Queue\ReliableQueueInterface;
+
+/**
+ * Redis queue implementation.
+ *
+ * @ingroup queue
+ */
+abstract class QueueBase implements ReliableQueueInterface {
+
+  /**
+   * The name of the queue this instance is working with.
+   *
+   * @var string
+   */
+  protected $name;
+
+  /**
+   * Key for list of available items.
+   *
+   * @var string
+   */
+  protected $availableListKey;
+
+  /**
+   * Key for list of claimed items.
+   *
+   * @var string
+   */
+  protected $claimedListKey;
+
+  /**
+   * Key prefix for items that are used to track expiration of leased items.
+   *
+   * @var string
+   */
+  protected $leasedKeyPrefix;
+
+  /**
+   * Key of increment counter key.
+   *
+   * @var string
+   */
+  protected $incrementCounterKey;
+
+  /**
+   * Key for hash table of available queue items.
+   *
+   * @var string
+   */
+  protected $availableItems;
+
+  /**
+   * Reserve timeout for blocking item claim.
+   *
+   * This will be set to number of seconds to wait for an item to be claimed.
+   * Non-blocking approach will be used when set to NULL.
+   *
+   * @var int|null
+   */
+  protected $reserveTimeout;
+
+  /**
+   * Constructs a \Drupal\Core\Queue\DatabaseQueue object.
+   *
+   * @param string $name
+   *   The name of the queue.
+   * @param array $settings
+   *   Array of Redis-related settings for this queue.
+   */
+  function __construct($name, array $settings) {
+    $this->name = $name;
+    $this->reserveTimeout = $settings['reserve_timeout'];
+    $this->availableListKey = 'drupal:queue:' . $name . ':avail';
+    $this->availableItems = 'drupal:queue:' . $name . ':items';
+    $this->claimedListKey = 'drupal:queue:' . $name . ':claimed';
+    $this->leasedKeyPrefix = 'drupal:queue:' . $name . ':lease:';
+    $this->incrementCounterKey = 'drupal:queue:' . $name . ':counter';
+  }
```

Comment:
There's probably a few methods from the Drupal\redis\Cache\CacheBase class that should be exported to either a trait or a base class, such as the getKey() method, instead of doing the concat by yourselves.

The same way, I'd prefer to have class constants instead of typing N times the same string, but that's just a matter of tests.